test/MusicStore.Test/CartSummaryComponentTest.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Session;
+using Microsoft.Framework.Caching.Distributed;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging.Testing;
+using MusicStore.Models;
+using Xunit;
+
+namespace MusicStore.Components
+{
+    public class CartSummaryComponentTest
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public CartSummaryComponentTest()
+        {
+            var services = new ServiceCollection();
+
+            services.AddEntityFramework()
+                      .AddInMemoryStore()
+                      .AddDbContext<MusicStoreContext>();
+
+            _serviceProvider = services.BuildServiceProvider();
+        }
+
+        [Fact]
+        public async Task CartSummaryComponent_Returns_CartedItems()
+        {
+            // Arrange
+            var viewContext = new ViewContext()
+            {
+                ViewData = new ViewDataDictionary(
+                    new EmptyModelMetadataProvider(),
+                    new ModelStateDictionary()),
+                HttpContext = new DefaultHttpContext()
+            };
+
+            // Session initialization
+            var cartId = "CartId_A";
+            var sessionFeature = new SessionFeature()
```

Comment:
I looked as Music Store app itself, and to my surprise, it doesn't seem to be using session state at all. It seems to be doing a "hand-rolled" session state by creating its own cookie and using that to store the cart id.

Does this unit test work just as well without all this session stuff? My guess is yes. However, that would also be surprising because that means that something else is missing here...